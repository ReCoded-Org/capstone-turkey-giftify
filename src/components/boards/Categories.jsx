import React from "react";
import "./Board.css";
import CardsByCategory from "./CardsByCategory";

function Categories({
  allCategories,
  setSelectedItems,
  allProducts,
  selectedItems,
  setSingleCard,
  setItemsDetails,
}) {
  return (
    
    allCategories.map((product, index) => {
    return (
      <div key={index} className="categories">
        <button
          key={index}
          className={`oneCategory ${product}`}
          onClick={(event) => {
            event.preventDefault();
            setSelectedItems(product);
          }}
        >
          {product}
        </button>
        {product === selectedItems ? (
          <CardsByCategory
            allProducts={allProducts}
            selectedItems={selectedItems}
            setSingleCard={setSingleCard}
            setItemsDetails={setItemsDetails}
          />
        ) : null}
      </div>
    );
  })
  );
}

export default Categories;
