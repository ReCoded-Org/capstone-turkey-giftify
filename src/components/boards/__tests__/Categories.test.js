import React from "react";
import Categories from "../Categories";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Categories rendered correctly", () => {
  const tree = renderer
    .create(<Categories allCategories />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
