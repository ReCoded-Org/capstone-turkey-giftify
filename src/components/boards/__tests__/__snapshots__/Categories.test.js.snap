// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories rendered correctly 1`] = `
Array [
  <div
    className="categories"
  >
    <button
      className="oneCategory book"
      onClick={[Function]}
    >
      book
    </button>
  </div>,
  <div
    className="categories"
  >
    <button
      className="oneCategory clothing"
      onClick={[Function]}
    >
      clothing
    </button>
  </div>,
]
`;
